src/passes/OptimizeInstructions.cpp

```diff
@@ -1224,11 +1230,113 @@ struct OptimizeInstructions
             !EffectAnalyzer(getPassOptions(), features, left->right)
                .hasSideEffects()) {
           switch (left->op) {
-            //   (x > y) | (x == y)    ==>    x >= y
             case EqInt32: {
-              if (right->op == GtSInt32) {
-                left->op = GeSInt32;
-                return left;
+              switch (right->op) {
+                //   (x == y) | (x >  y)   ==>    x >= y
+                //   (x == y) | (x >= y)   ==>    x >= y
+                case GtSInt32:
+                case GeSInt32: {
+                  left->op = GeSInt32;
+                  return left;
+                }
+                case GtUInt32:
+                case GeUInt32: {
+                  left->op = GeUInt32;
+                  return left;
+                }
+                //   (x == y) | (x  < y)    ==>    x <= y
+                //   (x == y) | (x <= y)    ==>    x <= y
+                case LtSInt32:
+                case LeSInt32: {
+                  left->op = LeSInt32;
+                  return left;
+                }
+                case LtUInt32:
+                case LeUInt32: {
+                  left->op = LeUInt32;
+                  return left;
+                }
+                //   (x == y) | (x != y)   ==>    1
+                case NeInt32: {
+                  return LiteralUtils::makeFromInt32(
+                    1, Type::i32, *getModule());
+                }
+                default: {
+                }
```

Comment:
I don't think this empty default case (or the ones below) should be necessary.

Reply:
Without this I can't compile due to `-Werror,-Wswitch`. But I'll use `default: break` instead